README.MD

```diff
@@ -12,14 +12,133 @@ The _jarvis-0.0.1-SNAPSHOT-spring-boot.jar_ is the previously mentioned JAR repa
 The Spring Boot Maven plugin also creates a copy of the Maven JAR named _jarvis-0.0.1-SNAPSHOT.jar.original_. 
 Run the application with this command.
 ```
-java -jar target/jarvis-0.0.1-SNAPSHOT-spring-boot.jar "arg1" "arg2"
+java -jar target/jarvis-0.0.1-SNAPSHOT-spring-boot.jar --context="arg1""arg1" "arg2"
 ```
 ---
 **NOTE**
 
 The name of the JAR will be the name in the pom.xml concatenated to the version from the pom.xml by a '-'.
 
 ---
+## Context
+A context represents how the test runner can reach local and external resources that are necessary for the tests. 
+Contexts are parsed from the __contexts.json__ file by the ContextLoader.
+The path to the __context.json__ is a required parameter (```--context```) to run the application.
+Each context has an id which correlates to the context's type shown in the examples below.
+Valid context types are BigQuery, MySQL, MSSQL, TalendAPI, SFTP and LocalContext.
+Beside id and type, the context can have various parameters depending on its type in key-value format.  
+Let's see a valid example for a context of each type.
+
+### BigQuery
+
+```
+[
+  {
+    "id": "{same as the folder name in the assert and pre folders}",
+    "contextType": "BigQuery",
+    "parameters": {
+      "project": "scoot-edw-dev",
+      "datasetNamePrefix": "csk_",
+      "edwCoreDataset": "EDW_CORE",
+      "stagingDataset": "STAGING"
+    }
+  }
+]
+```
+For the BigQuery type the _project_ is a required parameter, while _datasetNamePrefix_ is optional.  
+Any other parameters can be used as substitution variables in SQL files (when the key is used it is replaced by the value during query execution).  
```

Comment:
not just in SQL files, in any files used by the tests

Reply:
Done.